x-pack/plugins/security_solution/public/security_integrations/cribl/components/custom_cribl_form.tsx

```diff
@@ -0,0 +1,277 @@
+  index: number;
+  routeEntries: RouteEntry[];
+  datastreamOpts: string[];
+  onChangeCriblDataId(index: number, value: string): void;
+  onChangeDatastream(index: number, value: string): void;
+  onDeleteEntry(index: number): void;
+}
+
+const RouteEntryComponent = React.memo<RouteEntryComponentProps>(
+  ({
+    index,
+    routeEntries,
+    datastreamOpts,
+    onChangeCriblDataId,
+    onChangeDatastream,
+    onDeleteEntry,
+  }) => {
+    const routeEntry = routeEntries[index]; // the route entry for this row
+
+    const options = datastreamOpts.map((o) => ({
+      label: o,
+    }));
+
+    const selectedOption = options.filter((o) => o.label === routeEntry.datastream);
+
+    return (
+      <>
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFormRow label="Cribl _dataId field">
+              <EuiFieldText
+                value={routeEntry.dataId}
+                onChange={(e) => onChangeCriblDataId(index, e.currentTarget.value)}
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFormRow hasEmptyLabelSpace>
+            <EuiFlexItem grow={false}>
+              {i18n.translate('xpack.securitySolution.securityIntegration.cribl.mapsTo', {
+                defaultMessage: 'MAPS TO',
+              })}
+            </EuiFlexItem>
+          </EuiFormRow>
+          <EuiFlexItem>
+            <EuiFormRow label="Datastream">
+              <EuiComboBox
+                placeholder="Select"
+                singleSelection
+                options={options}
+                selectedOptions={selectedOption}
+                onChange={(o) => {
+                  if (o.length > 0) {
+                    onChangeDatastream(index, o[0].label);
+                  } else {
+                    onChangeDatastream(index, '');
+                  }
+                }}
+              />
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFormRow hasEmptyLabelSpace>
+            <EuiButtonIcon
+              color="danger"
+              iconType="trash"
+              onClick={() => onDeleteEntry(index)}
+              isDisabled={routeEntries.length === 1}
+              aria-label="entryDeleteButton"
+              className="itemEntryDeleteButton"
+              data-test-subj="itemEntryDeleteButton"
+            />
+          </EuiFormRow>
+        </EuiFlexGroup>
+      </>
+    );
+  }
+);
+
+export const CustomCriblForm = memo<PackagePolicyReplaceDefineStepExtensionComponentProps>(
+  ({ newPolicy, onChange, isEditPage }) => {
+    const { http } = useKibana().services;
+
+    const [missingReqPermissions, setMissingReqPermissions] = useState<boolean>();
+    const [datastreamOpts, setDatastreamOpts] = useState<string[]>([]);
+
+    const [initialPackagePolicy] = useState<NewPackagePolicy>(newPolicy);
+
+    useEffect(() => {
+      const fetchData = async () => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
For safety, consider `if (!http) return;` instead of the _no-non-null-assertion_ linter  exception

Reply:
Updated